### DISCUSSION TITLE
Im new to coding and need help with something

### DISCUSSION THREAD

**Discussion by lovedbyou** [nlohmann/json#4534] [Q&A] [answered]
How do i make a json file im very new to coding and need a json file for c++ and i cant figure out how to make one

**Comment by dingminghbut** [comment, marked as answer]
A JSON file is just a text file with JSON syntax. With this library, you can build the JSON value in C++ and write it to a file.

Small complete example:

```cpp
#include <fstream>
#include <iostream>
#include <nlohmann/json.hpp>

int main()
{
    nlohmann::json j;

    j["name"] = "Alice";
    j["age"] = 20;
    j["is_student"] = true;
    j["skills"] = {"C++", "JSON"};

    std::ofstream file("data.json");
    file << j.dump(4); // 4 spaces indentation

    std::cout << "Wrote data.json\n";
}
```

The created `data.json` will look like this:

```json
{
    "age": 20,
    "is_student": true,
    "name": "Alice",
    "skills": [
        "C++",
        "JSON"
    ]
}
```

To read it back later:

```cpp
#include <fstream>
#include <nlohmann/json.hpp>

int main()
{
    std::ifstream file("data.json");
    nlohmann::json j = nlohmann::json::parse(file);

    std::string name = j["name"];
}
```

If you are just starting, the important JSON shapes are:

- object: `{ "key": "value" }`
- array: `[1, 2, 3]`
- string: `"hello"`
- number: `123`
- boolean: `true` / `false`
- null: `null`
